src/watts/plugin_mcnp.py

```diff
@@ -1,15 +1,193 @@
 # SPDX-FileCopyrightText: 2022-2023 UChicago Argonne, LLC
 # SPDX-License-Identifier: MIT
 
-from typing import List, Optional
+import os
+import re
+from pathlib import Path
+from typing import List, Optional, Dict, Tuple
 
 from uncertainties import ufloat
 
+from .data import ATOMIC_SYMBOL, ATOMIC_NUMBER, isotopes
 from .fileutils import PathLike
 from .plugin import PluginGeneric, _find_executable
 from .results import Results
 
 
+def expand_element(xsdir: Optional[PathLike] = None):
+    def expand_element_inner(material: str, default_suffix: str = None) -> str:
+        """Expand elements in an MCNP material definition
+
+        Parameters
+        ----------
+        material
+            String representing material definition
+        default_suffix
+            Cross section suffix used by default if none is provided
+
+        Returns
+        -------
+        str
+            Material definition string with elements expanded
+
+        """
+        words = material.split()
+
+        # Note that the 'xsdir' variable is in an enclosing scope -- it gets
+        # passed in by PluginMCNP so that each instance of the plugin can
+        # uniquely set its own xsdir
+        available_nuclides = _get_nuclides_from_xsdir(xsdir)
+
+        # Determine whether filter block includes start of material card
+        # TODO: Use regex to break into multiple material blocks?
+        if len(words) % 2 == 1:
```

Comment:
should we address the TODO before moving forward?

Reply:
Yes, I can implement that as part of this PR.